File: codingame/community/executionCircle/executionCircle.scala
import math._
import scala.util._
object Solution extends App {
    val Array(n, s) = for(i <- readLine split " ") yield i.toLong
    val d = readLine
    var x:BigInt = 2;
    var answer:BigInt = 0;
    
    while (x * 2 < n) {
        x = x * 2;
    }

    if (d == "LEFT") {
        answer = s + (2 * (n - x));
    } else {
        answer = s - (2 * (n - x));
    }
    if (answer < 0) {
        answer = answer + n;
    }
    if (answer > n) {
        answer = answer - n;
    }
    println(answer)
}

File: codingame/community/executionCircle/executionCircle.lua

next_token = string.gmatch(io.read(), "[^%s]+")
n = tonumber(next_token())
s = tonumber(next_token())
d = io.read()

--print (n)
--print (s)
--print (d)

x = 2

while (x * 2) <= n do
    x = x * 2
end

--print (x)

if d == "LEFT"
    then answer = s + (2 * (n - x))
else
    answer = s - (2 * (n - x))
end

if answer < 0
   then answer = answer + n
end

if answer > n
   then answer = answer - n 
end

print (answer)
